pkg/pillar/cmd/zedrouter/networkinstance.go

```diff
@@ -332,12 +318,39 @@ func handleNetworkInstanceCreate(
 	ctx.networkInstanceStatusMap.Store(status.UUID, &status)
 	publishNetworkInstanceStatus(ctx, &status)
 
-	status.PInfo = make(map[string]types.ProbeInfo)
-	niUpdateNIprobing(ctx, &status)
+	// Select uplink interface for the network instance.
+	var selectedUplinkLL string
+	if config.WithUplinkProbing() {
+		probeStatus, err := ctx.uplinkProber.StartNIProbing(config)
+		if err != nil {
+			log.Errorf("doNetworkInstanceCreate(%s) failed: %v\n",
+				key, err)
+			log.Error(err)
+			status.SetErrorNow(err.Error())
+			status.ChangeInProgress = types.ChangeInProgressTypeNone
+			publishNetworkInstanceStatus(ctx, &status)
+			return
+		}
+		selectedUplinkLL = probeStatus.SelectedUplinkLL
+	} else {
+		selectedUplinkLL = config.Logicallabel
+	}
+	if selectedUplinkLL != "" {
+		ports := ctx.deviceNetworkStatus.GetPortsByLogicallabel(selectedUplinkLL)
+		if len(ports) == 1 {
+			status.SelectedUplinkIntf = ports[0].IfName
+		} else {
+			err := fmt.Errorf(
+				"label of selected uplink does not match single interface (%v)", ports)
+			log.Error(err)
+			status.SetErrorNow(err.Error())
```

Comment:
Shouldn't you return after the SetError?

Reply:
Yes, forgot to that - fixed, thanks for noticing.